core/field_colour.js

```diff
@@ -94,10 +95,11 @@ Blockly.FieldColour.prototype.getValue = function() {
 };
 
 /**
- * Set the colour.
+ * Set the colour. If opt_fromSliders is true, do not update the sliders.
  * @param {string} colour The new colour in '#rrggbb' format.
+ * @param {boolea} opt_fromSliders Flag to prevent sliders from recursing on themselves.
  */
-Blockly.FieldColour.prototype.setValue = function(colour) {
+Blockly.FieldColour.prototype.setValue = function(colour, opt_fromSliders) {
```

Comment:
Hm the problem is that updating the sliders causes a change event to fire, which would cause a loop, that is why I don't update the sliders if the update came from the sliders. Is there a different way to handle that?

Reply:
if the update came from the sliders, will the values be the same? in other words, rather than passing the isFromSliders into here, could you instead check if they value you're about to set them to is the same as what they already are and then short circuit that way?